### ISSUE TITLE
on-window-detected does not work when a window is revealed after being hidden

### ISSUE THREAD

**Issue by arglin** [nikitabobko/AeroSpace#425]
Thank you for the amazing work, but there is one thing that blocks me using aerospace. Here is my settings:
<img width="424" alt="image" src="https://github.com/user-attachments/assets/19de8f0b-366b-4ebc-8b79-1ce177967a7b">

so I use raycast to open Arc, it move the arc window to workspace b for the first time created. And somehow out of habit, I will use raycast to hide Arc. 
And then I switch the focus to workspace t in another monitor for other purpose. (now arc is hidden, and I am focuing on workspace t)
now I use raycast to open Arc agian, this time I guess because Arc window is already created, `on-window-detected` will not trigger. So the Arc will be  added to the workspace t (another monitor). 
This is not what I expected, because Arc window has changed to another workspace after reveal. 

Can we empower on-window-detected to support detection on the existing window when reveal, or add another callback like `on-window-reveal` to be able trigger commands to keep the window on certain workspace? 

## Checklist

- [x] I've searched for duplicates. My search queries were (please don't forget to search in closed issues too):
  - [query 1](is:issue on-window-detected is:closed)
  - [query 2](is:issue is:open on-window-detected)
  - ...
- [x] I've read the documentation https://nikitabobko.github.io/AeroSpace/guide & https://nikitabobko.github.io/AeroSpace/commands
- [x] I've searched in the documentation. My search queries were:
  - on-window-detected

```
aerospace CLI client version: 0.14.1-Beta 5861e85b65cb7dc914a0a68027c56e2bc90510fd
AeroSpace.app server version: 0.14.1-Beta 5861e85b65cb7dc914a0a68027c56e2bc90510fd
```

**Comment by arglin** [comment]
or another strategy is:
even the window is hidden, treat it as a window in the assigned workspace b, and when the window is revealed, even tho I am in another workspace t, aerospace will still display workspace b and focus on workspace b

**Comment by EliTheSurfer** [comment]
I have the exact same problem. When a hidden window reappears, it always appears in the currently focused window, which basically breaks the workspace/monitor assignment.

**Comment by crates51** [comment]
Yeap, i also have this problem with browser extensions
![image](https://github.com/user-attachments/assets/c39d4333-4736-46f8-a2f9-b78a0d28d4c2)

If i repopen the browser extension(by pressing nr 1 in image) then the extension popup(nr 2 in image) apears on my left screen bugged, even tho my browser is on my right screen

**Comment by nikitabobko** [comment]
> or another strategy is:
even the window is hidden, treat it as a window in the assigned workspace b, and when the window is revealed, even tho I am in another workspace t, aerospace will still display workspace b and focus on workspace b

I tend to agree with this. Also see: 
https://github.com/nikitabobko/AeroSpace/issues/18#issuecomment-1992634891
https://github.com/nikitabobko/AeroSpace/issues/18#issuecomment-2310013237

**Comment by kjvdven** [comment]
Came looking for this issue, when will this land?

I have the same with my terminal, which I toggle with a Raycast shortcut. Hope this will fix that too.

**Comment by nikitabobko** [comment]
Released in 0.15.0-Beta
